### PR TITLE
[vtadmin] [experimental] add per-api RBAC implementation

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This PR introduces an experimental feature to vtadmin, namely per-api scoped role-based access control. See the package documentation in rbac.go for more details around the principles behind the design. I plan to continue to iterate on this within our deployment, and I'll follow up with more unit tests and documentation as I do so.

## Checklist
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtadmin/api.go [hunk 1]
```diff
@@ -97,6 +101,38 @@ func NewAPI(clusters []*cluster.Cluster, opts Options) *API {
...
+	var (
+		authn rbac.Authenticator
+		authz *rbac.Authorizer
+	)
+	if opts.RBAC != nil {
+		authn = opts.RBAC.GetAuthenticator()
+		authz = opts.RBAC.GetAuthorizer()
+
+		if authn != nil {
+			opts.GRPCOpts.StreamInterceptors = append(opts.GRPCOpts.StreamInterceptors, rbac.AuthenticationStreamInterceptor(authn))
```
Comment (doeg) [inline go/vt/vtadmin/api.go:113]: I like dis. So clean.

File: go/vt/vtadmin/rbac/authentication.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+// Authenticator defines the interface vtadmin authentication plugins must
+// implement. Authenticators are installed at the grpc interceptor and http
+// middleware layers.
+type Authenticator interface {
+	// Authenticate returns an Actor given a context. This method is called
+	// from the stream and unary grpc server interceptors, and are passed the
+	// stream and request contexts, respectively.
+	//
+	// Returning an error from the authenticator will fail the request. To
+	// denote an authenticated request, return (nil, nil) instead.
```
Comment (doeg) [inline go/vt/vtadmin/rbac/authentication.go:43]: Such nice docs always ❤️

File: go/vt/vtadmin/rbac/rbac.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package rbac provides role-based access control for vtadmin API endpoints.
```
Comment (doeg) [inline go/vt/vtadmin/rbac/rbac.go:18]: chef kiss

**Review by doeg — APPROVED** [review]
Given that we already went over the feedback I had out-of-band + this works great internally, it looks good to me!

File: go/vt/vtadmin/rbac/authentication.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	Authenticate(ctx context.Context) (*Actor, error)
+	// AuthenticateHTTP returns an actor given an http.Request.
+	//
+	// Returning an error from the authenticator will fail the request. To
+	// denote an authenticated request, return (nil, nil) instead.
```
[the remark this answers is not in the document]
Comment (ajm188) [reply go/vt/vtadmin/rbac/authentication.go:48]: ohhh this is a typo. it should be "to denote an **un**authenticated request, return (nil, nil)"

**Pull request merged**